From b1e2aae0aa6edd91413b288b9cc6939f2d80e302 Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Tue, 17 Sep 2019 23:04:05 -0700
Subject: arm: avoid conditional branch to PLT in sigsetjmp

The R_ARM_THM_JUMP19 relocation type generated for the original code
when targeting Thumb 2 is not supported by the gold linker.

http://git.musl-libc.org/cgit/musl/commit/?id=b1e2aae0aa6edd91413b288b9cc6939f2d80e302

---
 src/signal/arm/sigsetjmp.s | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/signal/arm/sigsetjmp.s b/src/signal/arm/sigsetjmp.s
index 318addba..69ebbf49 100644
--- a/src/signal/arm/sigsetjmp.s
+++ b/src/signal/arm/sigsetjmp.s
@@ -6,9 +6,10 @@
 sigsetjmp:
 __sigsetjmp:
 	tst r1,r1
-	beq setjmp
+	bne 1f
+	b setjmp
 
-	str lr,[r0,#256]
+1:	str lr,[r0,#256]
 	str r4,[r0,#260+8]
 	mov r4,r0
 
-- 
cgit v1.2.1


